### DISCUSSION TITLE
Bulk rename Interfaces

### DISCUSSION THREAD

**Discussion by remchuk** [netbox-community/netbox#16932] [Q&A] [answered]
Hello there!

I would like to bulk rename the interfaces  across several devices.

Lets say I Have device type A with 300 devices accosciated with it. And I realize that I made a mistake and 
interface GigabitEthernet1/0/49 should be TenGigabitEthernet1/0/49. 
I really need to go one by one? That horrific.

The option for bulk rename shouldn't be associated to a single device at a time.

Is there  a feature for this? a plan for this? Any advice?

**Comment by a084ed22** [comment]
You can perhaps fix it on the DB side and then rebuild the cache.

**Comment by jeremystretch** [comment]
You can do this easily from the Python shell (`manage.py nbshell`) as a one-off:

```python
Interface.objects.filter(name='GigabitEthernet1/0/49').update(name='TenGigabitEthernet1/0/49')
```

Be aware however that this bulk approach will _not_ generate a changelog record. You may also need to filter by the assigned device type.

**Reply by remchuk** [reply]
Is my suggestion is not good ?

Why bulk renaming is exclusive to one device at a time?

**Reply by candlerb** [reply]
Because it would be very rarely needed. It's better to create your Device Types correctly in the first case, and validate them before generating hundreds of instances. There is a [Device Type library](https://github.com/netbox-community/devicetype-library) that you can copy from.

**Reply by Azmodeszer** [reply]
Well, you can just go to Interfaces, filter by your device type and then bulk-rename them that way. That's essentially what you're asking.

**Reply by remchuk** [reply]
> Well, you can just go to Interfaces, filter by your device type and then bulk-rename them that way. That's essentially what you're asking.

Actually you can't.

**Reply by Azmodeszer** [reply, marked as answer]
Devices > Interfaces > Filters > scroll down to Device type and select the model in question > Search > bulk select the interfaces you want to rename > Rename Selected > Find: "Giga" Replace: "TenGiga" > Preview > Apply

Unless we're talking about different things as to what you want to achieve?

**Reply by remchuk** [reply]
> Devices > Interfaces > Filters > scroll down to Device type and select the model in question > Search > bulk select the interfaces you want to rename > Rename Selected > Find: "Giga" Replace: "TenGiga" > Preview > Apply
> 
> Unless we're talking about different things as to what you want to achieve?

You are actually correct!

I'm using an older version, 3.5, and it is not possible there.
I just checked the demo.netbox with the latest version and it is an option!
